osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs

```diff
@@ -11,6 +11,11 @@ namespace osu.Game.Rulesets.Osu.Scoring
 {
     public class OsuScoreProcessor : ScoreProcessor
     {
+        public OsuScoreProcessor(Ruleset ruleset)
```

Comment:
This is a minor point but you could scope down these types like so:

```suggestion
        public OsuScoreProcessor(OsuRuleset ruleset)
```

to prevent anyone from wrongly constructing a frankenstein score processor with the entirely wrong ruleset.

Goes for all other similar changes in other rulesets, too.